tools/build_script_generator/cmake/resources/ci_CMakeLists.txt.in

```diff
@@ -0,0 +1,64 @@
+# Copyright (c) 2018, Sergiy Yevtushenko
+# Copyright (c) 2018-2019, Niklas Hauser
+# Copyright (c) 2010, Jacob Andersen
+#
+# This file is part of the modm project.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+# Autogenerated cmake file, should be replaced with your own cmake enviroment.
+# Can be turned off with Option(include_cmakelists) in project.xml
+#
+# Use: "make flash" to program your device with openocd after configuration of adapter in project.xml
+
+cmake_minimum_required(VERSION 3.15)
+set(CMAKE_SYSTEM_NAME Generic)
+set(CMAKE_SYSTEM_PROCESSOR arm)
+set(CMAKE_TRY_COMPILE_TARGET_TYPE STATIC_LIBRARY)
+%% if core.startswith("cortex-m")
+set(TOOLCHAIN_PREFIX "arm-none-eabi")
+%% elif core.startswith("avr")
+set(TOOLCHAIN_PREFIX "avr")
+%% else
+set(TOOLCHAIN_PREFIX "")
+%% endif
+set(CMAKE_C_COMPILER ${TOOLCHAIN_PREFIX}-gcc)
+set(CMAKE_ASM_COMPILER ${TOOLCHAIN_PREFIX}-as)
+set(CMAKE_CXX_COMPILER ${TOOLCHAIN_PREFIX}-g++)
+set(CMAKE_SIZE_UTIL ${TOOLCHAIN_PREFIX}-size)
+set(CMAKE_OBJCOPY ${TOOLCHAIN_PREFIX}-objcopy)
+set(CMAKE_OBJDUMP ${TOOLCHAIN_PREFIX}-objdump)
+set(CMAKE_AR ${TOOLCHAIN_PREFIX}-gcc-ar)
+set(CMAKE_NM ${TOOLCHAIN_PREFIX}-gcc-nm)
+
+project({{ project_name }} CXX C)
+
+if(NOT CMAKE_BUILD_TYPE)
+  set(CMAKE_BUILD_TYPE MinSizeRel)
+endif()
+
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+
+add_library(project_options INTERFACE)
+target_compile_features(project_options INTERFACE cxx_std_20)
+
+## Compile modm
+add_subdirectory(modm)
+
+## Compile example
+# We need to be compatible with examples. never do this, without reason.
+file(GLOB APP_SOURCE_FILES *.c *.cpp */*.c */*.cpp)
```

Comment:
Examples only have `*.cpp` files immediately in its project directory.

```suggestion
# We need to be compatible with examples. never do this, without reason.
file(GLOB APP_SOURCE_FILES *.cpp)
```

Reply:
No, a few of them have a folder for generated sources like XPCC or images. I got bitten by this bug too, hence one more kevel of globbing.